pkg/model/serializer.go

```diff
@@ -1185,6 +1206,45 @@ func (e *Environment) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// UnmarshalYAML Implements the Unmarshaler interface of the yaml pkg.
+func (ba *BuildArgs) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	buildArgs := make(BuildArgs, 0)
+	result, err := getBuildArgs(unmarshal)
+	if err != nil {
+		return err
+	}
+	for key, value := range result {
+		buildArgs = append(buildArgs, BuildArg{Name: key, Value: value})
+	}
+	sort.SliceStable(buildArgs, func(i, j int) bool {
+		return strings.Compare(buildArgs[i].Name, buildArgs[j].Name) < 0
+	})
+	*ba = buildArgs
+	return nil
+}
+
+func getBuildArgs(unmarshal func(interface{}) error) (map[string]string, error) {
+	result := make(map[string]string)
+
+	var rawList []BuildArg
+	err := unmarshal(&rawList)
+	if err == nil {
+		for _, label := range rawList {
```

Comment:
```suggestion
		for _, buildArg := range rawList {
```

I think it is more accurate than `label`, which in this context I think it doesn't mean anything relevant